modules/platforms/dotnet/Apache.Ignite.Core.Tests/Services/ServicesTest.cs

```diff
@@ -413,6 +413,50 @@ public void TestDuckTyping([Values(true, false)] bool local)
             Assert.IsInstanceOf<InvalidCastException>(ex.InnerException);
         }
 
+        /// <summary>
+        /// Test call service proxy from remote node with a methods having an array of user types and objects.
+        /// </summary>
+        [Test]
+        public void TestCallServiceProxyWithMethodsHavingArrays()
+        {
+            // Deploy to the remote node.
+            var nodeId = Grid2.GetCluster().GetLocalNode().Id;
+
+            var cluster = Grid1.GetCluster().ForNodeIds(nodeId);
+
+            cluster.GetServices().DeployNodeSingleton(SvcName, new TestIgniteServiceArraySerializable());
+
+            var typedArray = new[] {10, 11, 12}
+                .Select(x => new PlatformComputeBinarizable {Field = x}).ToArray();
+
+            var objArray = typedArray.ToArray<object>();
+
+            var prx = Services.GetServiceProxy<ITestIgniteServiceArray>(SvcName);
```

Comment:
Can we repeat these checks in "keep binary" mode as well? (with 
var binPrx = Services.WithKeepBinary().GetServiceProxy<ITestIgniteServiceArray>(SvcName);

Reply:
Done